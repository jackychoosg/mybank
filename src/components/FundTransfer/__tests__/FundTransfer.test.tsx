import { render, cleanup, fireEvent } from '@testing-library/react'
import { FundTransfer } from '../index'
describe('FundTransfer component', () => {
  afterEach(cleanup)
  const accounts = [
    {
      id: 1,
      name: 'Account 1',
      accountNumber: '111111111'
    },
    {
      id: 2,
      name: 'Account 2',
      accountNumber: '222222222'
    },
    {
      id: 3,
      name: 'Account 3',
      accountNumber: '333333333'
    }
  ]
  const currencies = [
    {
      id: 1,
      code: 'SGD',
      name: 'Singapore Dollar'
    },
    {
      id: 2,
      code: 'AUD',
      name: 'Australia Dollar'
    }
  ]
  const onDropdownChange = jest.fn()
  const onValueChange = jest.fn()
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <FundTransfer
        fromAccount={1}
        toAccount={2}
        accountDropdownOptions={accounts}
        amount={1000}
        onDropdownChange={onDropdownChange}
        onValueChange={onValueChange}
        description="Testing"
        currencies={currencies}
        currencyId={1}
        errors={{ amount: 'required' }}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should call onDropdownChange action', () => {
    const { getAllByTestId } = render(
      <FundTransfer
        fromAccount={1}
        toAccount={2}
        accountDropdownOptions={accounts}
        amount={1000}
        onDropdownChange={onDropdownChange}
        onValueChange={onValueChange}
        description="Testing"
        currencies={currencies}
        currencyId={1}
        errors={{}}
      />
    )
    const allDropdowns = getAllByTestId('select')
    const fromAccountDropdown = allDropdowns[0]
    const toAccountDropdown = allDropdowns[1]
    const currencyDropdown = allDropdowns[2]
    fireEvent.change(fromAccountDropdown, { target: { value: 2 } })
    fireEvent.change(toAccountDropdown, { target: { value: 1 } })
    fireEvent.change(currencyDropdown, { target: { value: 2 } })
    expect(onDropdownChange).toBeCalledTimes(3)
  })

  it('should call onValueChange action', () => {
    const { getByTestId } = render(
      <FundTransfer
        fromAccount={1}
        toAccount={2}
        accountDropdownOptions={accounts}
        amount={1000}
        onDropdownChange={onDropdownChange}
        onValueChange={onValueChange}
        description="Testing"
        currencies={currencies}
        currencyId={1}
        errors={{}}
      />
    )
    const textAreaIput = getByTestId('text-area')
    fireEvent.change(textAreaIput, { target: { value: 'description' } })
    expect(onValueChange).toHaveBeenCalled()
  })
})
