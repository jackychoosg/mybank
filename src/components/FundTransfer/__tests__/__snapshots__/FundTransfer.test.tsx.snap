// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FundTransfer component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM ivfALk"
  >
    <div
      class="sc-iCoGMd jbhMdW"
    >
      <label
        class="sc-fujyAs hTHZlC"
      >
        From account
      </label>
      <select
        class="sc-pNWdM FAkSM"
        data-testid="select"
      >
        <option
          class="sc-jrsJWt"
          data-testid="select-option"
          value="0"
        >
          ---Choose account---
        </option>
        <option
          class="sc-jrsJWt"
          data-testid="select-option"
          value="1"
        >
          Account 1
        </option>
        <option
          class="sc-jrsJWt"
          data-testid="select-option"
          value="3"
        >
          Account 3
        </option>
      </select>
    </div>
    <div
      class="sc-iCoGMd jbhMdW"
    >
      <label
        class="sc-fujyAs hTHZlC"
      >
        To account
      </label>
      <select
        class="sc-pNWdM FAkSM"
        data-testid="select"
      >
        <option
          class="sc-jrsJWt"
          data-testid="select-option"
          value="0"
        >
          ---Choose account---
        </option>
        <option
          class="sc-jrsJWt"
          data-testid="select-option"
          value="2"
        >
          Account 2
        </option>
        <option
          class="sc-jrsJWt"
          data-testid="select-option"
          value="3"
        >
          Account 3
        </option>
      </select>
    </div>
    <div
      class="sc-iCoGMd jbhMdW"
    >
      <label
        class="sc-fujyAs hTHZlC"
      >
        Currency
      </label>
      <select
        class="sc-pNWdM FAkSM"
        data-testid="select"
      >
        <option
          class="sc-jrsJWt"
          data-testid="select-option"
          value="0"
        >
          ---Choose currency---
        </option>
        <option
          class="sc-jrsJWt"
          data-testid="select-option"
          value="1"
        >
          Singapore Dollar
        </option>
        <option
          class="sc-jrsJWt"
          data-testid="select-option"
          value="2"
        >
          Australia Dollar
        </option>
      </select>
    </div>
    <div
      class="sc-gtsrHT jZTxQO"
    >
      <div
        class="sc-hKFxyN lgyOKa"
      >
        Amount
        <div
          class="sc-gKAaRy jfCmJa"
        >
          required
        </div>
      </div>
      <input
        class="sc-dlnjwi cpzDwW"
        error="error"
        id="input-amount"
        inputmode="decimal"
        name="amount"
        placeholder="Please enter amount"
        type="text"
        value="$1,000"
      />
    </div>
    <div
      class="sc-gtsrHT jZTxQO"
    >
      <div
        class="sc-hKFxyN lgyOKa"
      >
        Description(Optional)
      </div>
      <textarea
        class="sc-eCApnc dBBAYf"
        data-testid="text-area"
        maxlength="200"
      >
        Testing
      </textarea>
      <div
        class="sc-jSFjdj knYjhy"
      >
        193 characters remaining.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
